src/engraving/dom/dynamichairpingroup.cpp

```diff
@@ -235,18 +235,7 @@ void DynamicExpressionDragGroup::startDrag(EditData& ed)
 RectF DynamicExpressionDragGroup::drag(EditData& ed)
 {
     RectF r = static_cast<EngravingItem*>(m_dynamic)->drag(ed);
-
-    // Dynamic may snap to a different segment upon dragging,
-    // in which case move the expression with it
-    Segment* newSegment = m_dynamic->segment();
-    Segment* oldSegment = toSegment(m_expression->explicitParent());
-    staff_idx_t newStaff = m_dynamic->staffIdx();
-    staff_idx_t oldStaff = m_expression->staffIdx();
-
-    if (newSegment != oldSegment || newStaff != oldStaff) {
-        Score* score = newSegment->score();
-        score->undoChangeParent(m_expression, newSegment, newStaff);
-    }
+    r.unite(static_cast<EngravingItem*>(m_expression)->drag(ed));
```

Comment:
What is the purpose of the `static_cast<EngravingItem*>`? (This means _not_ the same as 
```c++
    RectF r = m_dynamic->EngravingItem::drag(ed);
    r.unite(m_expression->EngravingItem::drag(ed));
```
)